backend/src/users/schemas/user.schema.ts

```diff
@@ -0,0 +1,57 @@
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { Types, HydratedDocument } from 'mongoose';
+import * as bcrypt from 'bcrypt'
+
+export type UserDocument = HydratedDocument<User> & {
+  
+};
+
+@Schema()
+export class User {
+  @Prop({required: true})
+  name: string;
+
+  @Prop({required: true, unique: true})
+  email: string;
+
+  @Prop({select: false}) //
+  password?: string;
+
+  @Prop({select: false})
+  googleId?: string;
+
+  // association
+  @Prop({default: []})
+  friends: Types.ObjectId[];
+
+  @Prop({default: []})
+  groups: Types.ObjectId[];
+
+  @Prop({default: []})
+  expenses: Types.ObjectId[];
+
+  @Prop({default: []})
+  receipts: Types.ObjectId[];
+}
+
+export const UserSchema = SchemaFactory.createForClass(User);
+
+
+UserSchema.pre<UserDocument>('save', async function (next) {
```

Comment:
I noticed that in the auth service, when you create/register a user, you manually call bcrypt to hash their password. If we're adding a pre method to hash a user's password, we may not need to manually call it within the service. May want to look at the redundancy, but I like the idea of having a pre method here